src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Integer.Signed.D.cs

```diff
@@ -13,64 +13,64 @@ private static bool TryParseSByteD(ReadOnlySpan<byte> text, out sbyte value, out
 
             int sign = 1;
             int index = 0;
-            int num = text[index];
-            if (num == '-')
+            uint digit = text[index] - 48u;  // '0'
+            if (digit == unchecked('-' - 48u)) // '0'
             {
                 sign = -1;
                 index++;
                 if ((uint)index >= (uint)text.Length)
                     goto FalseExit;
-                num = text[index];
+                digit = text[index] - 48u; // '0'
             }
-            else if (num == '+')
+            else if (digit == unchecked('+' - 48u)) // '0'
             {
                 index++;
                 if ((uint)index >= (uint)text.Length)
                     goto FalseExit;
-                num = text[index];
+                digit = text[index] - 48u; // '0'
             }
 
-            int answer = 0;
+            uint answer = 0;
 
-            if (ParserHelpers.IsDigit(num))
+            if (digit <= 9)
             {
-                if (num == '0')
+                if (digit == 0)
                 {
                     do
                     {
                         index++;
                         if ((uint)index >= (uint)text.Length)
                             goto Done;
-                        num = text[index];
-                    } while (num == '0');
-                    if (!ParserHelpers.IsDigit(num))
+                        digit = text[index];
+                    } while (digit == '0');
+
+                    digit = digit - 48u; // '0'
+                    if (digit > 9)
                         goto Done;
                 }
 
-                answer = num - '0';
+                answer = digit;
                 index++;
 
                 if ((uint)index >= (uint)text.Length)
                     goto Done;
-                num = text[index];
-                if (!ParserHelpers.IsDigit(num))
+                digit = text[index] - 48u; // '0';
+                if (digit > 9)
                     goto Done;
                 index++;
-                answer = 10 * answer + num - '0';
+                answer = 10 * answer + digit;
 
                 // Potential overflow
                 if ((uint)index >= (uint)text.Length)
                     goto Done;
-                num = text[index];
-                if (!ParserHelpers.IsDigit(num))
+                digit = text[index] - 48u; // '0';
+                if (digit > 9)
                     goto Done;
-                if (answer > SByte.MaxValue / 10 + 1)
-                    goto FalseExit; // Overflow
-                answer = answer * 10 + num - '0';
+                answer = 10 * answer + digit;
 
                 // if sign < 0, (-1 * sign + 1) / 2 = 1
                 // else, (-1 * sign + 1) / 2 = 0
-                if ((uint)answer > (uint)SByte.MaxValue + (-1 * sign + 1) / 2)
+                if (answer > (uint)sbyte.MaxValue + (-1 * sign + 1) / 2)
                     goto FalseExit; // Overflow
                 index++;
```

Comment:
nit: move `index++` above `answer = 10 * answer + digit` to be consistent with other changes (like in TryParseInt32D [here](https://github.com/dotnet/corefx/pull/25130/files#diff-5e90497df430f6541664f0a780acac46R317))

Reply:
Moved.